pkg/steps/multi_stage.go

```diff
@@ -0,0 +1,202 @@
+package steps
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	coreapi "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	coreclientset "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/junit"
+)
+
+const (
+	multiStageTestLabel = "ci.openshift.io/multi-stage-test"
+)
+
+type multiStageTestStep struct {
+	name            string
+	config          *api.ReleaseBuildConfiguration
+	podClient       PodClient
+	jobSpec         *api.JobSpec
+	pre, test, post []api.TestStep
+	subTests        []*junit.TestCase
+}
+
+func MultiStageTestStep(
+	testConfig api.TestStepConfiguration,
+	config *api.ReleaseBuildConfiguration,
+	podClient PodClient,
+	jobSpec *api.JobSpec,
+) (api.Step, error) {
+	return newMultiStageTestStep(testConfig, config, podClient, jobSpec), nil
```

Comment:
why we need an extra unexported function and pass again the same args there, to return a new struct?

Also, this doesn't return any error.

Reply:
Error removed, the extra function is used by the tests.